tools/benchmark/src/Reporter.ts

```diff
@@ -281,21 +264,13 @@ export class BenchmarkReporter {
 
 	private writeCompletedBenchmarks(
 		suiteName: string,
-		benchmarks: Map<string, BenchmarkResult>,
+		benchmarks: Map<string, Readonly<BenchmarkResult>>,
 	): string {
-		const outputFriendlyBenchmarks: unknown[] = [];
-
-		for (const [key, bench] of benchmarks.entries()) {
-			if (!isResultError(bench)) {
-				// successful benchmarks: ready them for output to file
-				outputFriendlyBenchmarks.push(this.outputFriendlyObjectFromBenchmark(key, bench));
-			}
-		}
-
 		// Use the suite name as a filename, but first replace non-alphanumerics with underscores
 		const suiteNameEscaped: string = suiteName.replace(/[^\da-z]/gi, "_");
+		const benchmarksArray: Readonly<BenchmarkResult>[] = [...benchmarks.values()];
 		const outputContentString: string = JSON.stringify(
-			{ suiteName, benchmarks: outputFriendlyBenchmarks },
+			{ suiteName, benchmarksArray },
```

Comment:
A couple of things:
- I think we're losing the `benchmarkName` property that used to exist in the output, which is important. I don't think there's any other way to know which test corresponds to a given entry in the array.
- Let's keep the same name for the property, `benchmarks` instead of `benchmarksArray`. The parameter name here is not a problem, we can still use "benchmarks" as a key when stringifying.
- The previous logic skipped entries for tests that errored out; I think we want to keep that.

Reply:
* I can add the `testName` into `customData`, previously memory tests have this property but general tests don't. I can add it then all type of test will have `testName`.
* Revert back to `benchmarks`.
* Skip error here.

Reply:
Latest Sample report:
~~~
{
    "suiteName": "Matrix memory usage Detached Matrix",
    "benchmarks": [
        {
            "elapsedSeconds": 6.57157636,
            "customData": {
                "Heap Used Avg": 1755.2842105263157,
                "Margin of Error": 41.43532399006374,
                "Relative Margin of Error": 2.360604837756702,
                "Heap Used StdDev": 460.7458927692714,
                "Iterations": 500,
                "testName": "@Benchmark @Measurement @MemoryUsage Create empty Matrix"
            },
            "cu...
~~~